File: ios/ArchiveSample/ArchiveSample/ViewController.m
//
//  ViewController.m
//  ArchiveSample
//
//  Created by fujita taisuke on 2012/08/03.
//  Copyright (c) 2012年 Revolution 9. All rights reserved.
//

#import "ViewController.h"
#import "Person.h"
#import "Address.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
}

- (void)viewDidUnload
{
    [super viewDidUnload];
    // Release any retained subviews of the main view.
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPhone) {
        return (interfaceOrientation != UIInterfaceOrientationPortraitUpsideDown);
    } else {
        return YES;
    }
}

- (IBAction)respondToSaveButtonClick:(id)sender {
    /*
    NSArray *paths = NSSearchPathForDirectoriesInDomains(
                                                         NSDocumentDirectory,
                                                         NSUserDomainMask, YES);
    NSLog(@"%@", paths[0]);
    */
    
    NSString *directory = [NSHomeDirectory() stringByAppendingPathComponent:@"Documents"];
    NSString *filePath = [directory stringByAppendingPathComponent:@"data.dat"];
    
    Person *tYamada = [[Person alloc] init];
    tYamada.name = @"山田太郎";
    Address *yAddress = [[Address alloc] init];
    yAddress.zipCode = @"104-0061";
    yAddress.state = @"東京都";
    yAddress.city = @"中央区";
    yAddress.other = @"銀座1丁目";
    tYamada.address = yAddress;
    
    Person *hYamada = [[Person alloc] init];
    hYamada.name = @"山田花子";
    hYamada.address = yAddress;
    
    Person *tanaka = [[Person alloc] init];
    tanaka.name = @"田中次郎";
    Address *tAddress = [[Address alloc] init];
    tAddress.zipCode = @"145-0071";
    tAddress.state = @"東京都";
    tAddress.city = @"大田区";
    tAddress.other = @"田園調布1丁目";
    tanaka.address = tAddress;
    NSArray *array = @[tYamada, hYamada, tanaka];
    BOOL successful = [NSKeyedArchiver archiveRootObject:array toFile:filePath];
    if (successful) {
        NSLog(@"%@", @"データの保存に成功しました。");
    }
    /*
     NSArray *array = @[@"山田太郎", @"東京都中央区"];
     BOOL successful = [NSKeyedArchiver archiveRootObject:array toFile:filePath];
     if (successful) {
     NSLog(@"%@", @"データの保存に成功しました。");
     }
     */
}

- (IBAction)respondToLoadButtonClick:(id)sender {
    NSString *directory = [NSHomeDirectory() stringByAppendingPathComponent:@"Documents"];
    NSString *filePath = [directory stringByAppendingPathComponent:@"data.dat"];
    NSArray *array = [NSKeyedUnarchiver unarchiveObjectWithFile:filePath];
    if (array) {
        for (Person *person in array) {
            NSLog(@"%@", person.name);
            NSLog(@"%@", person.address.zipCode);
            NSLog(@"%@", person.address.state);
            NSLog(@"%@", person.address.city);
            NSLog(@"%@", person.address.other);
        }
    } else {
        NSLog(@"%@", @"データが存在しません。");
    }
    /*
     NSArray *array = [NSKeyedUnarchiver unarchiveObjectWithFile:filePath];
     if (array) {
     for (NSString *data in array) {
     NSLog(@"%@", data);
     }
     } else {
     NSLog(@"%@", @"データが存在しません。");
     }
     */
}

@end

File: ios_swift/Archive/Archive/ViewController.swift
//
//  ViewController.swift
//  Archive
//
//  Created by taisuke fujita on 2014/07/07.
//  Copyright (c) 2014年 taisuke fujita. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
                            
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func respondToArchiveButton() {
        // ファイルパスの取得
        let paths = NSSearchPathForDirectoriesInDomains(
            .DocumentDirectory,
            .UserDomainMask, true) as! [String]
        // 保存するファイルの名前
        //let filePath = String(paths[0]) + "data.dat"
        let filePath = String(paths[0]).stringByAppendingPathComponent("data.plist")
        // 保存するデータ、氏名と住所
        let array = ["山田太郎", "104-0061", "東京都", "中央区", "銀座1丁目"]
        // アーカイブしてdata.datというファイル名で保存する
        let successful = NSKeyedArchiver.archiveRootObject(array, toFile: filePath)
        if successful {
            println("データの保存に成功しました。")
        }
    }
    
    @IBAction func respondToUnarchiveButton() {
        let paths = NSSearchPathForDirectoriesInDomains(
            .DocumentDirectory,
            .UserDomainMask, true) as! Array<String>
        //let filePath = String(paths[0]) + "data.dat"
        let filePath = String(paths[0]).stringByAppendingPathComponent("data.plist")
        if let array = NSKeyedUnarchiver.unarchiveObjectWithFile(filePath) as? Array<String> {
            for str in array {
                println(str)
            }
        } else {
            println("データがありません")
        }
    }
}

